Client/Frontend/Browser/HandoffManager.swift

```diff
@@ -0,0 +1,45 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+class HandoffManager: NSObject {
+    private static var _sharedInstance: HandoffManager?
```

Comment:
There is no need to have a private var and a public one, you can just create the singleton instance with

`static var sharedInstance = HandoffManager()`